---
title: "GeneCoEx"
author: "Chenxin Li"
date: "11/9/2022"
output:  
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Packages 
```{r}
library(tidyverse)
library(Seurat) 
library(Signac)
library(readxl)

library(igraph)
library(ggraph)

library(RColorBrewer)
library(viridis)
library(rcartocolor)

library(patchwork)

set.seed(666)
```


# Data
## Seurat object and cell type annotation 
```{r}
CRO_RNA <- read_rds("../Results/R_output/CRO_RNA.Rds")

leaf_cell_type_assignment <- data.frame(
  cluster = c(0:15)
) %>% 
  mutate(cell_type = case_when(
    cluster == 0 | 
      cluster == 1 | 
      cluster == 3 | 
      cluster ==7  ~ "Mesophyll",
    cluster == 2 | 
      cluster == 5 ~ "Epidermis",
    cluster == 12 ~ "Guard cells",
    cluster == 11 |
      cluster == 6 |
      cluster == 10 |
      cluster == 14 |
      cluster == 15  ~ "Vasculature",
    cluster == 13 ~ "IPAP",
      cluster == 8 ~ "Idioblast",
    T ~ "Unassigned"
  )) %>% 
  mutate(cell_type = factor(cell_type, 
                            levels = c(
                              "Mesophyll", "Epidermis", "Guard cells",
                              "Vasculature", 
                              "IPAP", "Idioblast", "Unassigned"
                            ))) %>% 
 mutate(cluster = factor(cluster, levels = c(
    "0", "1", "3", "7", 
    "2","5", 
    "12",
    "11", "6", "10", "14", "15",
    "13", "8",
    "4", "9"
  )))

Leaf_cell_type_strip <- leaf_cell_type_assignment %>% 
  ggplot(aes(x = cluster, y = "" )) +
  geom_tile(aes(fill = cell_type)) +
  scale_fill_manual(values = c(brewer.pal(6, "Accent"), "grey80")) +
  labs(fill = "Cell type") +
  theme_void() +
  theme(
    legend.position = "bottom" ,
    text = element_text(size = 14)
  )

CRO_RNA
```

## MIA enzymes 
```{r}
leaf_MIA_genes <- read_csv("../Results/R_output/leaf_MIA_genes.csv")
head(leaf_MIA_genes)
```

## Functional annotation 
```{r}
functional_annoation <- read_delim("../Data/cro_v3_anno/cro_v3.functional_annotation.txt",
                                   delim = "\t", col_names = F)

functional_annoation <- functional_annoation %>% 
  mutate(gene_ID = str_sub(X1, end = 13)) %>% 
  distinct(gene_ID, .keep_all = T) %>% 
  mutate(gene_ID = str_replace(gene_ID, "_", "-"))

tail(functional_annoation)
```

## Gene models
```{r}
repr_gene_models <- read_delim("../Data/cro_v3_anno/cro_v3.gene_models.repr.gff3",
                               delim = "\t", col_names = F)

repr_loci <- repr_gene_models %>% 
  dplyr::filter(X3 == "mRNA") %>% 
  dplyr::select(X1, X4, X5, X7, X9) %>% 
  separate(X9, c("ID", "Name"), sep = ";") %>% 
  dplyr::select(-Name) %>% 
  separate(ID, c("temp", "gene_ID"), sep = "=") %>% 
  dplyr::select(-temp) %>% 
  mutate(LocusID = str_sub(gene_ID, start = 1, end = 13)) %>% 
  mutate(gene_ID = str_replace(gene_ID, "_", "-")) %>% 
  mutate(LocusID = str_replace(LocusID, "_", "-")) %>% 
  dplyr::rename(
    Chr = X1,
    start = X4,
    end = X5,
    strand = X7
  )


head(repr_loci)
```


## Known TFs 
```{r}
MIA_genes_info <- readRDS(file = "../Data/MIA_genes_info.Rds")

MIA_TFs <- MIA_genes_info %>% 
  dplyr::filter(segment == "TF") %>% 
  mutate(gene = str_replace(gene, "_", "-")) 

MIA_TFs
```
```{r}
write_excel_csv(MIA_TFs, "../Results/R_output/MIA_TFs.csv")
```


# Pull gene expression matrix 
```{r}
CRO_RNA@meta.data <- CRO_RNA@meta.data %>% 
  mutate(barcode = row.names(CRO_RNA@meta.data))
```


```{r}
CRO_gex <- GetAssayData(CRO_RNA, slot = "scale.data", assay = "integrated")

CRO_gex_by_cluster <- CRO_gex %>% 
  as.data.frame() %>% 
  mutate(gene = row.names(.)) %>% 
  pivot_longer(cols = !gene, names_to = "barcode", values_to = "exp") %>% 
  inner_join(CRO_RNA@meta.data, by = "barcode") %>% 
  group_by(gene, seurat_clusters) %>% 
  summarise(mean.rel.exp = mean(exp)) %>% 
  ungroup()

head(CRO_gex_by_cluster)
```

# Pairwise correlation 
```{r}
z_score_wide <- CRO_gex_by_cluster %>% 
  pivot_wider(names_from = seurat_clusters, values_from = mean.rel.exp) %>% 
  as.data.frame()

row.names(z_score_wide) <- z_score_wide$gene
head(z_score_wide)[1:6]
```

```{r}
z_score_wide %>% 
  inner_join(leaf_MIA_genes, by = c("gene"="gene_ID")) %>% 
  dplyr::select(gene, tag, order2) %>% 
  arrange(order2)
```
32 of 41 known steps are in the top 3000 variable genes 

```{r}
cor_matrix <- cor(t(z_score_wide[, -1]))
dim(cor_matrix)
```

## Edge selection 
```{r}
number_of_clusters <- ncol(z_score_wide) - 1
cor_matrix_upper_tri <- cor_matrix
cor_matrix_upper_tri[lower.tri(cor_matrix_upper_tri)] <- NA
```


```{r}
edge_table <- cor_matrix_upper_tri %>% 
  as.data.frame() %>% 
  mutate(from = row.names(cor_matrix)) %>% 
  pivot_longer(cols = !from, names_to = "to", values_to = "r") %>% 
  dplyr::filter(is.na(r) == F) %>% 
  dplyr::filter(from != to) %>% 
  mutate(t = r*sqrt((number_of_clusters-2)/(1-r^2))) %>% 
  mutate(p.value = case_when(
    t > 0 ~ pt(t, df = number_of_clusters-2, lower.tail = F),
    t <=0 ~ pt(t, df = number_of_clusters-2, lower.tail = T)
  )) %>% 
  mutate(FDR = p.adjust(p.value, method = "fdr")) 

head(edge_table)
```

## Check bait gene correlation 
```{r}
## IPAP - ISY and IO
edge_table %>% 
  dplyr::filter(str_detect(from, "CRO-07G007680|CRO-03G004950") &
                  str_detect(to, "CRO-07G007680|CRO-03G004950")) 

```
```{r}
# Epidermis LAMT, SLS, SLTr
edge_table %>% 
  dplyr::filter(str_detect(from, "CRO-03G019430|CRO-04G013510|CRO-03G032350") &
                  str_detect(to, "CRO-03G019430|CRO-04G013510|CRO-03G032350")) 
```
```{r}
# Epidermis Redox1 and Redox2 
edge_table %>% 
  dplyr::filter(str_detect(from, "CRO-05G017150|CRO-05G008760") &
                  str_detect(to, "CRO-05G017150|CRO-05G008760")) 
```
```{r}
## Idioblast - D4H and DAT
edge_table %>% 
  dplyr::filter(str_detect(from, "CRO-06G013360|CRO-02G001090|CRO-06G024620") &
                  str_detect(to, "CRO-06G013360|CRO-02G001090|CRO-06G024620")) 
```

```{r}
edge_table %>% 
  dplyr::filter(r > 0) %>% 
  dplyr::filter(FDR < 0.05) %>% 
  slice_min(order_by = abs(r), n = 10)

edge_table %>% 
  dplyr::filter(r > 0) %>% 
  dplyr::filter(FDR < 0.01) %>% 
  slice_min(order_by = abs(r), n = 10)
```

Use 0.75 as cutoff

```{r}
edge_table_select <- edge_table %>% 
  dplyr::filter(r >= 0.75) %>% 
  mutate(from = str_sub(from, end = 13)) %>% 
  mutate(to = str_sub(to, end = 13)) 

dim(edge_table_select)
```

# Module detection
## Node table 
```{r}
colnames(leaf_MIA_genes)
colnames(MIA_TFs)
```
```{r}
known_genes <- leaf_MIA_genes %>% 
  mutate(gene_ID = str_sub(gene_ID, end = 13)) %>% 
  dplyr::select(gene_ID, tag, order, segment) %>% 
  rbind(
    MIA_TFs %>% 
      mutate(gene_ID = gene) %>% 
      dplyr::select(gene_ID, tag, order, segment)
  )
```


```{r}
node_table <- data.frame(
  gene_ID = unique(c(edge_table_select$from, edge_table_select$to))
) %>% 
  mutate(gene_ID = str_sub(gene_ID, end = 13)) %>% 
  left_join(functional_annoation, by = "gene_ID") %>% 
  dplyr::rename(functional_annoation = X2) %>% 
  left_join(known_genes, by = "gene_ID") %>% 
  mutate(segment = case_when(
    tag == "DPAS" ~ "Alkaloid",
    T ~ segment
  ))

node_table %>% 
  dplyr::filter(is.na(tag) == F) %>% 
  arrange(order)

dim(node_table)
```

## Network object
```{r}
my_network <- graph_from_data_frame(
  edge_table_select,
  vertices = node_table,
  directed = F
)
```

### Optimize resolution
```{r}
optimize_resolution <- function(network, resolution){
  modules = network %>% 
    cluster_leiden(resolution_parameter = resolution,
                   objective_function = "modularity")
  
  parsed_modules = data.frame(
    gene_ID = names(membership(modules)),
    module = as.vector(membership(modules)) 
    )
  
  num_module_5 = parsed_modules %>% 
    group_by(module) %>% 
    count() %>% 
    arrange(-n) %>% 
    dplyr::filter(n >= 5) %>% 
    nrow() %>% 
    as.numeric()
  
  num_genes_contained = parsed_modules %>% 
    group_by(module) %>% 
    count() %>% 
    arrange(-n) %>% 
    dplyr::filter(n >= 5) %>% 
    ungroup() %>% 
    summarise(sum = sum(n)) %>% 
    as.numeric()
  
  cbind(num_module_5, num_genes_contained) %>% 
    as.data.frame()

}

optimize_resolution(my_network, 1)
```

```{r}
 optimization_results <- purrr::map_dfr(
  .x = seq(from = 0.25, to = 5, by = 0.25),
  .f = optimize_resolution, 
  network = my_network
) %>% 
  cbind(
   resolution = seq(from = 0.25, to = 5, by = 0.25)
  )  

head(optimization_results)
```

```{r}
Optimize_num_module <- optimization_results %>% 
  ggplot(aes(x = resolution, y = num_module_5)) +
  geom_line(size = 1.1, alpha = 0.8, color = "dodgerblue2") +
  geom_point(size = 3, alpha = 0.7) +
  geom_vline(xintercept = 1.25, size = 1, linetype = 4) +
  labs(x = "resolution parameter",
       y = "num. modules\nw/ >=5 genes") +
  theme_classic() +
  theme(
    text = element_text(size = 14),
    axis.text = element_text(color = "black")
  )

Optimize_num_gene <- optimization_results %>% 
  ggplot(aes(x = resolution, y = num_genes_contained)) +
  geom_line(size = 1.1, alpha = 0.8, color = "violetred2") +
  geom_point(size = 3, alpha = 0.7) +
  geom_vline(xintercept = 1.25, size = 1, linetype = 4) +
  labs(x = "resolution parameter",
       y = "num. genes in\nmodules w/ >=5 genes") +
  theme_classic() +
  theme(
    text = element_text(size = 14),
    axis.text = element_text(color = "black")
  )

wrap_plots(Optimize_num_module, Optimize_num_gene, nrow = 2)
```

# Graph based clustering 
```{r}
modules <- cluster_leiden(my_network, resolution_parameter = 4, 
                          objective_function = "modularity")

my_network_modules <- data.frame(
  gene_ID = names(membership(modules)),
  module = as.vector(membership(modules)) 
) %>% 
  inner_join(node_table, by = "gene_ID")

module_5 <- my_network_modules %>% 
  group_by(module) %>% 
  count() %>% 
  arrange(-n) %>% 
  dplyr::filter(n >= 5)

my_network_modules <- my_network_modules %>% 
  dplyr::filter(module %in% module_5$module)

head(my_network_modules)
```
## Module treatment correspondance
```{r}
z_score_modules <- CRO_gex_by_cluster %>% 
  mutate(gene_ID = str_sub(gene, end = 13)) %>% 
  inner_join(my_network_modules %>% 
               dplyr::select(-X1), by = "gene_ID") 

head(z_score_modules)
```
```{r}
modules_mean_z <- z_score_modules %>% 
  group_by(module, seurat_clusters) %>% 
  summarise(mean.z = mean(mean.rel.exp),
            q3.z = quantile(mean.rel.exp, 0.75)) %>% 
  ungroup()

head(modules_mean_z)
```

```{r}
module_peak_exp <- modules_mean_z %>% 
  group_by(module) %>% 
  slice_max(order_by = mean.z, n = 1) %>% 
  dplyr::rename(peak_seurat_cluster = seurat_clusters) %>% 
  arrange(peak_seurat_cluster)

module_peak_exp
```

```{r}
module_peak_exp %>% 
  filter(peak_seurat_cluster == 2 |
           peak_seurat_cluster == 5 |
           peak_seurat_cluster == 8 |
           peak_seurat_cluster == 13)
```


```{r}
modules_mean_z %>% 
  group_by(module) %>% 
  slice_min(order_by = mean.z, n = 1) %>% 
  dplyr::rename(min_seurat_cluster = seurat_clusters) %>% 
  arrange(min_seurat_cluster)
```

## Line graphs 
### Positive modules 
```{r}
module_line_plot <- z_score_modules %>% 
  mutate(seurat_clusters = factor(seurat_clusters, levels = c(
   "0", "1", "3", "7", 
    "2","5", 
    "12",
    "11", "6", "10", "14", "15",
    "13", "8",
    "4", "9"
  ))) %>% 
  dplyr::filter(module == "5" |
                  module == "6" |
                  module == "8" ) %>% 
  ggplot(aes(x = seurat_clusters, y = mean.rel.exp)) +
  facet_grid(module ~. , scales = "free_y") +
  geom_line(aes(group = gene_ID), alpha = 0.3, color = "grey75") +
  geom_line(
    data = z_score_modules %>%
      dplyr::filter(gene_ID == "CRO-07G007680" |
                  gene_ID == "CRO-04G013510" |
                  gene_ID == "CRO-02G001090") %>%
      mutate(seurat_clusters = factor(seurat_clusters, levels = c(
        "0", "1", "3", "7", 
    "2","5", 
    "12",
    "11", "6", "10", "14", "15",
    "13", "8",
    "4", "9"
        ))),
    size = 1.1, alpha = 0.8, aes(color= tag, group = module)
  ) +
  scale_color_manual(values = carto_pal(7, "BluGrn")[c(3, 5, 7)],
                     limits = c("ISY", "SLS1", "DAT")) +
  labs(x = "cell cluster",
       y = "z score",
       color = "example") +
  theme_classic() +
  theme(
    text = element_text(size = 14),
    axis.text = element_text(color = "black"),
    panel.spacing = unit(1, "line"),
    legend.position = "top",
    strip.text.y.right = element_blank()
  )

wrap_plots(module_line_plot, Leaf_cell_type_strip,
           nrow = 2, heights = c(1, 0.08))

ggsave("../Results/R_output/Module_line.svg", height = 4.5, width = 6.5, bg = "white")
ggsave("../Results/R_output/Module_line.png", height = 4.5, width = 6.5, bg = "white")
```

# Network graph 

```{r}
subnetwork_genes <- my_network_modules %>% 
  dplyr::filter(module == "5" |
           module == "6" |
           module == "8") %>% 
  mutate(segment = case_when(
    str_detect(segment, "Transporter") ~ "Alkaloid", 
    T ~ segment
    )) %>% 
  mutate(tag2 = case_when(
    is.na(tag) ~ "",
    T ~ "known genes"
    )) 

subnetwork_genes %>% 
  dplyr::filter(is.na(tag) ==F)

subnetwork_edges <- edge_table_select %>% 
  dplyr::filter(from %in% subnetwork_genes$gene_ID & 
                  to %in% subnetwork_genes$gene_ID)

dim(subnetwork_genes)
dim(subnetwork_edges) 

dim(subnetwork_genes)[1] == length(unique(c(subnetwork_edges$from, subnetwork_edges$to)))

write_excel_csv(subnetwork_genes, "../Results/R_output/subnetwork_genes.csv")
write_excel_csv(subnetwork_edges, "../Results/R_output/subnetwork_edges.csv")
```

```{r}
my_subnetwork <- graph_from_data_frame(subnetwork_edges,
                                     vertices = subnetwork_genes,
                                     directed = F)
```


```{r}
 my_subnetwork %>% 
  ggraph(
    layout = "kk", 
    circular = F) +
  geom_edge_diagonal(color = "grey80", width = 0.5, alpha = 0.5) +
  geom_node_point(alpha = 0.8, color = "white", shape = 21,
                  aes(fill = segment, size = tag2)) + 
  geom_node_text(aes(label = tag), size = 3, repel = T, max.overlaps = 30) +
  scale_fill_manual(values = c(viridis(5), "white"),
                   limits = c( "TF", "MEP", "Iridoid",
                             "Alkaloid", "Late Alkaloid", "")) +
  scale_size_manual(values = c(1, 2.5),
                    limits = c("", "known genes")) +
  labs(fill = "Stage of pathway") +
  guides(size = "none",
         fill = guide_legend(override.aes = list(size = 4), 
                             title.position = "top", nrow = 3)) +
  theme_void()+
  theme(
    text = element_text(size = 14), 
    legend.position = "bottom",
    legend.justification = 1,
    title = element_text(size = 12)
  )

ggsave("../Results/R_output/subnetwork_graph.svg", height = 5, width = 5, bg = "white")
ggsave("../Results/R_output/subnetwork_graph.png", height = 5, width = 5, bg = "white")
```

# Write out results 
```{r}
subnetwork_genes %>% 
  arrange(module) %>% 
  mutate(Module = case_when(
    module == "5" ~ "Epidermis",
    module == "6" ~ "Idioblast",
    module == "8" ~ "IPAP"
  )) %>% 
  write_excel_csv("../Results/R_output/Gene_CoExp_Module.csv")
```

## Graph some candidates 
### Dot plot 
```{r}
DefaultAssay(CRO_RNA) <- "RNA"

idioblast_redox <- subnetwork_genes %>% 
  dplyr::filter(module == "8") %>% 
  dplyr::filter(str_detect(functional_annoation, 
                           "erox|ytochro|2OG|elicitor")) %>% 
  mutate(class = case_when(
    str_detect(functional_annoation, "450") ~ "P450",
    str_detect(functional_annoation, "erox") ~ "Peroxidase",
    str_detect(functional_annoation, "2OG") ~ "2OG",
    T ~ "ADH"
  )) %>% 
  arrange(class) %>% 
  left_join(repr_loci, by = c("gene_ID" = "LocusID"))

idioblast_redox
```
```{r}
Candidate_genes <- DotPlot(CRO_RNA, features = idioblast_redox$gene_ID.y, group.by = "cell_type")

Candidate_dot <- Candidate_genes$data %>% 
  mutate(id2 = case_when(
    str_detect(id, "Meso") ~ "M",
    str_detect(id, "Epi") ~ "E",
    str_detect(id, "IP") ~ "IP",
    str_detect(id, "Idio") ~ "I",
    str_detect(id, "Vas") ~ "V",
    str_detect(id, "Guard") ~ "GC",
    str_detect(id, "Un") ~ "Un"
  )) %>% 
  mutate(id2 = factor(id2, levels = c(
    "IP", "E", "I",
    "M", "V", "GC", "Un"
  ))) %>% 
  mutate(internal.tag = row.names(.)) %>% 
  arrange(internal.tag) %>% 
  mutate(gene_ID = case_when(
    is.na(features.plot) ~ str_remove_all(internal.tag, "rna_"),
    T ~ features.plot %>% as.character()
  )) %>% 
  mutate(gene_ID = str_sub(gene_ID, start = 1, end = 15)) %>% 
  inner_join(idioblast_redox, by = c("gene_ID" = "gene_ID.y")) %>%
  ggplot(aes(y = gene_ID, x = id2)) +
  facet_grid(class ~., space = "free_y", scales = "free_y") +
  geom_point(aes(size = pct.exp, fill = avg.exp.scaled), shape = 21, color = "white") +
  # scale_y_discrete(label = NULL) +
  scale_fill_viridis(option = "A", begin = 0, end = 0.9) +
  guides(size = guide_legend(override.aes = list(fill = "grey40"),
                             title.position = "top", nrow = 2)) +
  guides(fill = guide_colorbar(title.position = "top")) +
  labs(x = NULL,
       y = NULL, 
       fill = "Avg. Exp.",
       size = "% Exp.") +
  #theme_minimal() +
  theme_classic() +
  theme(text = element_text(size = 14, color = "black"),
        axis.text = element_text(color = "black"),
        axis.text.y = element_text(hjust = 0.5),
        axis.text.x = element_text(hjust = 1, angle = 45, face = "plain"),
        legend.position = "top",
        legend.box = "horizontal", 
        legend.key.height = unit(0.8, "lines"),
        legend.key.width = unit(0.8, "lines"),
        panel.spacing = unit(1, "lines"), 
        title = element_text(size = 10, face = "bold"),
        strip.text = element_blank(),
        strip.background = element_blank()) 

Candidate_color_strip <- idioblast_redox %>% 
  ggplot(aes(x = "", y = gene_ID.y)) +
  facet_grid(class ~., space = "free_y", scales = "free_y") +
  geom_tile(aes(fill = class)) +
  scale_fill_manual(values = carto_pal(4, "Vivid")) +
  labs(fill = NULL) +
  guides(fill = guide_legend(ncol = 1), title.position = "top") +
  theme_void() +
  theme(
    text = element_text(size = 14, color = "black"),
    legend.key.height = unit(0.8, "lines"),
    legend.key.width = unit(0.8, "lines"),
    panel.spacing = unit(1, "lines"),
    strip.text = element_blank(),
    legend.position = "right"
  )

Candidate_labels <- idioblast_redox %>% 
  mutate(txt = case_when(
    str_detect(gene_ID.y, "01G033230") ~ "THAS1",
    is.na(tag) ~ " ",
    T ~ tag
  )) %>% 
  ggplot(aes(x = "", y = gene_ID.y)) +
  facet_grid(class ~., space = "free_y", scales = "free_y") +
  geom_text(aes(label = txt)) +
  theme_void() +
  theme(
    text = element_text(size = 14, color = "black"),
    legend.key.height = unit(0.8, "lines"),
    legend.key.width = unit(0.8, "lines"),
    panel.spacing = unit(1, "lines"),
    strip.text = element_blank(),
    legend.position = "right"
  )

wrap_plots(Candidate_dot, Candidate_color_strip, Candidate_labels,
           nrow = 1, widths = c(1, 0.05, 0.5), guides = "collect") &
  theme(legend.position = "right")

ggsave("../Results/R_output/Idioblast_candidates.svg", height = 6, width = 5, bg = "white")
ggsave("../Results/R_output/Idioblast_candidates.png", height = 6, width = 5, bg = "white")
```
### Line plot
```{r}
idioblast_redox
z_score_modules
```

```{r}
candidate_line_plot <- z_score_modules %>% 
  dplyr::filter(gene %in% idioblast_redox$gene_ID.y) %>% 
  left_join(idioblast_redox %>% 
              dplyr::select(gene_ID.y, class), 
            by = c("gene" = "gene_ID.y")) %>% 
  mutate(seurat_clusters = factor(seurat_clusters, levels = c(
  "0", "1", "3", "7", 
    "2","5", 
    "12",
    "11", "6", "10", "14", "15",
    "13", "8",
    "4", "9"
  ))) %>% 
  ggplot(aes(x = seurat_clusters, y = mean.rel.exp)) +
  facet_wrap( ~ gene, scales = "free_y", ncol = 3) +
  geom_line(aes(group = gene, color = class), alpha = 0.8, size = 1.1) +
  scale_color_manual(values = carto_pal(4, "Vivid")) +
  labs(x = "Seurat cluster",
       y = "z score",
       color = NULL) +
  theme_classic() +
  theme(
    text = element_text(size = 14),
    axis.text = element_text(color = "black"),
    panel.spacing = unit(1, "line"),
    legend.position = "top",
    strip.text.y = element_text(angle = 0),
    strip.background = element_blank()
  )

candidate_line_plot

ggsave("../Results/R_output/Idioblast_candidates_line.svg", height = 8, width = 10, bg = "white")
ggsave("../Results/R_output/Idioblast_candidates_line.png", height = 8, width = 10, bg = "white")
```

# Compare modules from protoplast 
## Pull membership from protoplast scRNA-seq
```{r}
protoplast_modules <- read_excel("../Data/Gene_coexp_modules.xlsx")

protoplast_modules <- protoplast_modules %>% 
  dplyr::rename(protoplast_module = module,
                gene_ID = gene)

head(protoplast_modules)
```

## Find neighbors of known genes 
```{r}
temp1 <- edge_table_select %>% 
  dplyr::filter(from %in% known_genes$gene_ID |
           to %in% known_genes$gene_ID)  

temp2 <- unique(c(temp1$from, temp1$to))
```



```{r}
subnetwork_genes_small <- subnetwork_genes %>% 
  mutate(nuclei_module = case_when(
    module == "5" ~ "Epidermis",
    module == "6" ~ "Idioblast",
    module == "8" ~ "IPAP"
  )) %>% 
  dplyr::filter(gene_ID %in% temp2)

head(subnetwork_genes_small)

subnetwork_genes_small %>% 
  group_by(nuclei_module) %>% 
  count()
```
## Write function
```{r}
find_jaccard <- function(cell_type_a, cell_type_b){
  
  nuclei = subnetwork_genes_small %>% 
    dplyr::filter(nuclei_module == cell_type_a) %>% 
    distinct(gene_ID) %>% 
    dplyr::filter(str_detect(gene_ID, ".S") == F)
  
  protoplast = protoplast_modules %>% 
    dplyr::filter(protoplast_module == cell_type_b) %>% 
    distinct(gene_ID) %>% 
    dplyr::filter(str_detect(gene_ID, ".S") == F)
  
  I = dplyr::intersect(nuclei$gene_ID, protoplast$gene_ID) %>% length()
  U = unique(c(nuclei$gene_ID, protoplast$gene_ID)) %>% length()
  
  n_nuclei = nrow(nuclei)
  n_protoplast = nrow(protoplast)
  
  cbind(I, U, n_nuclei, n_protoplast) %>% 
    as.data.frame() %>% 
    mutate(jaccard = I/U)
}

find_jaccard("IPAP", "IPAP")
```

## Apply
```{r}
cell_types_combo <- base::expand.grid(
  cell_type_a = c("IPAP", "Epidermis", "Idioblast"),
  cell_type_b = c("IPAP", "Epidermis", "Idioblast")
)

membership_overlap <- map2_dfr(.f = find_jaccard,
                             .x = cell_types_combo$cell_type_a, 
                             .y = cell_types_combo$cell_type_b) %>% 
  cbind(cell_types_combo) 
   

head(membership_overlap, 10)
```

## Graph
```{r}
membership_overlap %>% 
  mutate(cell_type_b = fct_rev(cell_type_b)) %>% 
  mutate(txt_col = case_when(
    jaccard > 0.15 ~ "white",
    T ~ "black"
  )) %>% 
  mutate(cell_type_a2 = str_sub(cell_type_a, end = 2)) %>% 
  mutate(cell_type_a2 = factor(cell_type_a2, levels = c(
    "IP", "Ep", "Id"
  ))) %>% 
  ggplot(aes(x = cell_type_a2, y = cell_type_b)) +
  geom_tile(aes(fill = jaccard)) +
  geom_text(aes(label = I, color = txt_col), fontface = "bold") +
  scale_fill_gradientn(colors = carto_pal(7, "PurpOr"),
                       breaks = c(0, 0.125, 0.25)) +
  scale_color_identity() +
  labs(x = "nuclei\nco-exp. module",
       y = "protoplast\nco-exp. module") +
  theme_classic() +
  theme(
    text = element_text(size = 14),
    axis.text = element_text(color = "black"),
    axis.text.y = element_text(hjust = 0.5)
  ) +
  coord_fixed()

ggsave("../Results/R_output/module_jaccard.svg", height = 3.5, width = 3.7, bg = "white")
ggsave("../Results/R_output/module_jaccard.png", height = 3.5, width = 3.7, bg = "white")
```

# Fancier graphs 
```{r}
CRO_gex_by_cell_type <- CRO_gex %>% 
  as.data.frame() %>% 
  mutate(gene = row.names(.)) %>% 
  pivot_longer(cols = !gene, names_to = "barcode", values_to = "exp") %>% 
  inner_join(CRO_RNA@meta.data, by = "barcode") %>% 
  group_by(gene, cell_type) %>% 
  summarise(mean.rel.exp = mean(exp)) %>% 
  ungroup() 

head(CRO_gex_by_cell_type)

write_excel_csv(CRO_gex_by_cell_type, "../Results/R_output/CRO_gex_by_cell_type.csv")
```

```{r}
my_subnetwork2 <- graph_from_data_frame(subnetwork_edges,
                                     vertices = subnetwork_genes %>% 
                                       left_join(
                                         CRO_gex_by_cell_type %>% 
                                           pivot_wider(names_from = "cell_type", values_from = mean.rel.exp) %>% 
                                           mutate(gene_ID = str_sub(gene, end = 13)),
                                         by = "gene_ID"
                                       ),
                                     directed = F)

```

## IPAP 
```{r}
 my_subnetwork2 %>% 
  ggraph(layout = "kk", circular = F) +
  geom_edge_diagonal(color = "grey80", width = 0.5, alpha = 0.5) +
  geom_node_point(alpha = 0.8, color = "white", shape = 21,
                  aes(fill = IPAP, size = tag2)) + 
  geom_node_text(aes(label = tag), size = 3, repel = T, max.overlaps = 30) +
  scale_fill_gradientn(colors = brewer.pal(9, "YlGnBu"),
                       breaks = c(0, 6),
                       labels = c("low", "high")) +
  scale_size_manual(values = c(1, 2.5),
                    limits = c("", "known genes")) +
  labs(fill = "Exp. at IPAP") +
  guides(size = "none",
         fill = guide_colorbar(title.position = "top")) +
  theme_void()+
  theme(
    text = element_text(size = 14), 
    legend.position = "bottom",
    title = element_text(size = 12)
  )

ggsave("../Results/R_output/subnetwork_graph_IPAP.svg", height = 5, width = 5, bg = "white")
ggsave("../Results/R_output/subnetwork_graph_IPAP.png", height = 5, width = 5, bg = "white")
```
## Epidermis
```{r}
 my_subnetwork2 %>% 
  ggraph(layout = "kk", circular = F) +
  geom_edge_diagonal(color = "grey80", width = 0.5, alpha = 0.5) +
  geom_node_point(alpha = 0.8, color = "white", shape = 21,
                  aes(fill = Epidermis, size = tag2)) + 
  geom_node_text(aes(label = tag), size = 3, repel = T, max.overlaps = 30) +
  scale_fill_gradientn(colors = brewer.pal(9, "PuBuGn"),
                       breaks = c(0, 1.5),
                       labels = c("low", "high")) +
  scale_size_manual(values = c(1, 2.5),
                    limits = c("", "known genes")) +
  labs(fill = "Exp. at epidermis") +
  guides(size = "none",
         fill = guide_colorbar(title.position = "top")) +
  theme_void()+
  theme(
    text = element_text(size = 14), 
    legend.position = "bottom",
    title = element_text(size = 12)
  )
ggsave("../Results/R_output/subnetwork_graph_Epi.svg", height = 5, width = 5, bg = "white")
ggsave("../Results/R_output/subnetwork_graph_Epi.png", height = 5, width = 5, bg = "white")
```
## Idioblast
```{r}
my_subnetwork2 %>% 
  ggraph(layout = "kk", circular = F) +
  geom_edge_diagonal(color = "grey80", width = 0.5, alpha = 0.5) +
  geom_node_point(alpha = 0.8, color = "white", shape = 21,
                  aes(fill = Idioblast, size = tag2)) + 
  geom_node_text(aes(label = tag), size = 3, repel = T, max.overlaps = 30) +
  scale_fill_gradientn(colors = brewer.pal(9, "PuRd"),
                       breaks = c(0, 2),
                       labels = c("low", "high")) +
  scale_size_manual(values = c(1, 3),
                    limits = c("", "known genes")) +
  labs(fill = "Exp. at idoblast") +
  guides(size = "none",
         fill = guide_colorbar(title.position = "top")) +
  theme_void()+
  theme(
    text = element_text(size = 14), 
    legend.position = "bottom",
    title = element_text(size = 12)
  )

ggsave("../Results/R_output/subnetwork_graph_Id.svg", height = 5, width = 5, bg = "white")
ggsave("../Results/R_output/subnetwork_graph_Id.png", height = 5, width = 5, bg = "white")
```

# TFs 
```{r}
MYB_TFs <- subnetwork_genes %>% 
  dplyr::filter(str_detect(functional_annoation, "myb|MYB")) %>% 
  left_join(repr_loci, by = c("gene_ID" = "LocusID")) %>% 
  arrange(module)

Homeo_TF <- subnetwork_genes %>% 
  dplyr::filter(str_detect(functional_annoation, "homeo|Homeo")) %>% 
  left_join(repr_loci, by = c("gene_ID" = "LocusID")) %>% 
  arrange(module)

WRKY_TF <- subnetwork_genes %>% 
  dplyr::filter(str_detect(functional_annoation, "WRKY|wrky")) %>% 
  left_join(repr_loci, by = c("gene_ID" = "LocusID")) %>% 
  arrange(module)

WRKY_TF
```

## MYB
```{r}
#DefaultAssay(CRO_multiome) <- "RNA"

DotPlot(CRO_RNA, features = MYB_TFs$gene_ID.y, 
        group.by = "cell_type")$data %>% 
  mutate(id2 = case_when(
    str_detect(id, "Meso") ~ "M",
    str_detect(id, "Epi") ~ "E",
    str_detect(id, "IP") ~ "IP",
    str_detect(id, "Idio") ~ "I",
    str_detect(id, "Vas") ~ "V",
    str_detect(id, "Guard") ~ "GC",
    str_detect(id, "Un") ~ "Un"
  )) %>% 
  mutate(id2 = factor(id2, levels = c(
    "IP", "E", "I",
    "M", "V", "GC", "Un"
  ))) %>% 
  inner_join(
    MYB_TFs, by = c("features.plot"="gene_ID.y")
  ) %>% 
  mutate(order_y = case_when(
    module == 8 ~ 1,
    module == 5 ~ 2,
    module == 6 ~ 3
  )) %>% 
  mutate(features.plot = reorder(features.plot, order_y)) %>% 
  ggplot(aes(y = features.plot, x = id2)) +
  geom_point(aes(size = pct.exp, fill = avg.exp.scaled), shape = 21, color = "white") +
  # scale_y_discrete(label = NULL) +
  scale_fill_viridis(option = "A", begin = 0, end = 0.9) +
  scale_size(breaks = c(10, 25, 50)) +
  guides(size = guide_legend(override.aes = list(fill = "grey40"),
                             title.position = "top", ncol = 1)) +
  guides(fill = guide_colorbar(title.position = "top", order = 1)) +
  labs(x = NULL,
       y = NULL, 
       fill = "Avg.\nExp.",
       size = "% Exp.",
       title = "MYB TFs in modules") +
  #theme_minimal() +
  theme_classic() +
  theme(text = element_text(size = 14, color = "black"),
        axis.text = element_text(color = "black"),
        axis.text.y = element_text(hjust = 0.5),
        axis.text.x = element_text(hjust = 1, angle = 45, face = "plain"),
        legend.position = "right",
        legend.box = "vertical",
        legend.key.height = unit(0.8, "lines"),
        legend.key.width = unit(0.8, "lines"),
        legend.title = element_text(size = 12),
        title = element_text(size = 10, face = "plain")) 

ggsave("../Results/R_output/MYB_TFs.svg", height = 4, width = 4, bg = "white")
ggsave("../Results/R_output/MYB_TFs.png", height = 4, width = 4, bg = "white")
```
## Homeo 
```{r}
DotPlot(CRO_RNA, features = Homeo_TF$gene_ID.y, 
        group.by = "cell_type")$data %>% 
  mutate(id2 = case_when(
    str_detect(id, "Meso") ~ "M",
    str_detect(id, "Epi") ~ "E",
    str_detect(id, "IP") ~ "IP",
    str_detect(id, "Idio") ~ "I",
    str_detect(id, "Vas") ~ "V",
    str_detect(id, "Guard") ~ "GC",
    str_detect(id, "Un") ~ "Un"
  )) %>% 
  mutate(id2 = factor(id2, levels = c(
    "IP", "E", "I",
    "M", "V", "GC", "Un"
  ))) %>% 
  inner_join(
    Homeo_TF, by = c("features.plot"="gene_ID.y")
  ) %>% 
  mutate(order_y = case_when(
   module == 8 ~ 1,
    module == 5 ~ 2,
    module == 6 ~ 3
  )) %>% 
  mutate(features.plot = reorder(features.plot, -order_y)) %>% 
  #dplyr::filter(module == 6) %>% 
  ggplot(aes(y = features.plot, x = id2)) +
  geom_point(aes(size = pct.exp, fill = avg.exp.scaled), shape = 21, color = "white") +
  # scale_y_discrete(label = NULL) +
  scale_fill_viridis(option = "A", begin = 0, end = 0.9) +
  guides(size = guide_legend(override.aes = list(fill = "grey40"),
                             title.position = "top", ncol = 1)) +
  guides(fill = guide_colorbar(title.position = "top", order = 1)) +
  labs(x = NULL,
       y = NULL, 
       fill = "Avg.\nExp.",
       size = "% Exp.",
       title = "Homeodomain TFs in modules") +
  #theme_minimal() +
  theme_classic() +
  theme(text = element_text(size = 14, color = "black"),
        axis.text = element_text(color = "black"),
        axis.text.y = element_text(hjust = 0.5),
        axis.text.x = element_text(hjust = 1, angle = 45, face = "plain"),
        legend.position = "right",
        legend.box = "vertical",
        legend.key.height = unit(0.8, "lines"),
        legend.key.width = unit(0.8, "lines"),
        legend.title = element_text(size = 12),
        title = element_text(size = 10, face = "plain")) 

ggsave("../Results/R_output/Homeo_TFs.svg", height = 4, width = 4, bg = "white")
ggsave("../Results/R_output/Homeo_TFs.png", height = 4, width = 4, bg = "white")
```
## WRKY
```{r}
DotPlot(CRO_RNA, features = WRKY_TF$gene_ID.y, 
        group.by = "cell_type")$data %>% 
  mutate(id2 = case_when(
    str_detect(id, "Meso") ~ "M",
    str_detect(id, "Epi") ~ "E",
    str_detect(id, "IP") ~ "IP",
    str_detect(id, "Idio") ~ "I",
    str_detect(id, "Vas") ~ "V",
    str_detect(id, "Guard") ~ "GC",
    str_detect(id, "Un") ~ "Un"
  )) %>% 
  mutate(id2 = factor(id2, levels = c(
    "IP", "E", "I",
    "M", "V", "GC", "Un"
  ))) %>% 
  inner_join(
    WRKY_TF, by = c("features.plot"="gene_ID.y")
  ) %>% 
  mutate(order_y = case_when(
    module == 8 ~ 1,
    module == 5 ~ 2,
    module == 6 ~ 3
  )) %>% 
  mutate(features.plot = reorder(features.plot, order_y)) %>% 
  ggplot(aes(y = features.plot, x = id2)) +
  geom_point(aes(size = pct.exp, fill = avg.exp.scaled), shape = 21, color = "white") +
  # scale_y_discrete(label = NULL) +
  scale_fill_viridis(option = "A", begin = 0, end = 0.9) +
  scale_size(breaks = c(10, 25, 50)) + 
  guides(size = guide_legend(override.aes = list(fill = "grey40"),
                             title.position = "top", ncol = 1)) +
  guides(fill = guide_colorbar(title.position = "top", order = 1)) +
  labs(x = NULL,
       y = NULL, 
       fill = "Avg.\nExp.",
       size = "% Exp.",
       title = "WRKY TFs in modules") +
  #theme_minimal() +
  theme_classic() +
  theme(text = element_text(size = 14, color = "black"),
        axis.text = element_text(color = "black"),
        axis.text.y = element_text(hjust = 0.5),
        axis.text.x = element_text(hjust = 1, angle = 45, face = "plain"),
        legend.position = "right",
        legend.box = "vertical",
        legend.key.height = unit(0.8, "lines"),
        legend.key.width = unit(0.8, "lines"),
        legend.title = element_text(size = 12),
        title = element_text(size = 10, face = "plain")) 

ggsave("../Results/R_output/WRKY_TFs.svg", height = 4, width = 4, bg = "white")
ggsave("../Results/R_output/WRKY_TFs.png", height = 4, width = 4, bg = "white")
```
## GATA
```{r}
GATA_heat <- DotPlot(CRO_RNA, features = c("CRO-03G006290.2"))$data %>% 
  ggplot(aes(y = features.plot, x = id)) +
  geom_point(aes(size = pct.exp, fill = avg.exp.scaled), shape = 21, color = "white") +
  # scale_y_discrete(label = NULL) +
  scale_fill_viridis(option = "A", begin = 0, end = 0.9) +
  guides(size = guide_legend(override.aes = list(fill = "grey40"),
                             title.position = "top", ncol = 1)) +
  guides(fill = guide_colorbar(title.position = "top", order = 1)) +
  labs(x = NULL,
       y = NULL, 
       fill = "Avg.\nExp.",
       size = "% Exp.",
       title = "CrGATA1 expression in leaf") +
  #theme_minimal() +
  theme_classic() +
  theme(axis.text.y = element_text(hjust = 0.5),
        #axis.text.x = element_text(hjust = 1, angle = 45, face = "plain"),
        legend.position = "right",
        legend.box = "vertical",
        legend.key.height = unit(0.8, "lines"),
        legend.key.width = unit(0.8, "lines"),
        title = element_text(size = 10)) 

wrap_plots(GATA_heat, Leaf_cell_type_strip,
           nrow = 2, heights = c(1, 0.1))

ggsave("../Results/R_output/GATA1_TF.svg", height = 3, width = 6.8, bg = "white")
ggsave("../Results/R_output/GATA1_TF.png", height = 3, width = 6.8, bg = "white")
```

