---
title: "JASPAR"
author: "Chenxin Li"
date: "11/16/2022"
output:  
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Packages
## New ones 
```{r}
# BiocManager::install("JASPAR2020")
# BiocManager::install("TFBSTools")
# BiocManager::install("motifmatchr")
# BiocManager::install("chromVAR")
```

```{r}
library(tidyverse)
library(Seurat) 
library(Signac)
library(readxl)

library(JASPAR2020)
library(TFBSTools)

library(BSgenome)

library(RColorBrewer)
library(viridis)
library(rcartocolor)

library(patchwork)

set.seed(666)
```

# Forge CROv3 genomes
```{r}
# c(paste0("Chr", 1:8), "ChrChl", paste0("scf_", 1:180))
```

```{r}
# forgeBSgenomeDataPkg("BSgenome.CRO_v3.UGA.CL-seed")
```

```{r}
# detach("package:BSgenome.CRO.UCSC.v3", unload=TRUE)
library(BSgenome.CRO.UCSC.v3)

# BSgenome.CRO.UCSC.v3
head(seqnames(BSgenome.CRO.UCSC.v3), 12)
unique(rownames(CRO_multiome) %>% str_sub(end = 4)) 
```

# Data 
```{r}
CRO_multiome <- readRDS("../Results/R_output/CRO_multiome.Rds")
DefaultAssay(CRO_multiome)
```

# Add motif info to Seurat object
```{r}
pfm <- getMatrixSet(
  x = JASPAR2020,
  opts = list(collection = "CORE", tax_group = 'plants', all_versions = FALSE)
)

main.chroms <- standardChromosomes(BSgenome.CRO.UCSC.v3) 
keep.peaks <- which(as.character(seqnames(granges(CRO_multiome))) %in% main.chroms)

CRO_multiome[["MACS2"]] <- subset(CRO_multiome[["MACS2"]], 
                                  features = rownames(CRO_multiome[["MACS2"]])[keep.peaks])

# add motif information
CRO_multiome <- AddMotifs(
  object = CRO_multiome,
  genome = BSgenome.CRO.UCSC.v3,
  pfm = pfm,
  assay = "MACS2"
)

CRO_multiome
```


```{r}
DefaultAssay(CRO_multiome) <- "MACS2"

da_peaks_Idioblast <- FindMarkers(
  object = CRO_multiome,
  ident.1 = "Idioblast",
  only.pos = T,
  test.use = "LR",
  min.pct = 0.05, 
  latent.vars = 'nCount_peaks',
  group.by = "cell_type"
)

da_peaks_IPAP <- FindMarkers(
  object = CRO_multiome,
  ident.1 = "IPAP",
  only.pos = T,
  test.use = "LR",
  min.pct = 0.05, 
  latent.vars = 'nCount_peaks',
  group.by = "cell_type"
)

da_peaks_Epi <- FindMarkers(
  object = CRO_multiome,
  ident.1 = "Epidermis",
  only.pos = T,
  test.use = "LR",
  min.pct = 0.05, 
  latent.vars = 'nCount_peaks',
  group.by = "cell_type"
)

top.idioblat.peaks <- rownames(da_peaks_Idioblast[da_peaks_Idioblast$p_val_adj < 0.05, ])
top.IPAP.peaks<- rownames(da_peaks_IPAP[da_peaks_IPAP$p_val_adj < 0.05, ])
top.epi.peaks <- rownames(da_peaks_Epi[da_peaks_Epi$p_val_adj < 0.05, ])
```

## Check D4H peak
```{r}
top.idioblat.peaks %>% 
  as.data.frame() %>% 
  dplyr::rename(peak = ".") %>% 
  separate(peak, c("Chr", "start", "end"), sep = "-", remove = F) %>% 
  arrange(Chr, start, end) %>% 
  dplyr::filter(Chr == "Chr2") %>% 
  dplyr::filter(start > 1605477 - 5000) %>% 
  dplyr::filter(end < 1606353 + 5000) 
```
 
## Check 16OMT-T16H2 peaks 
```{r}
top.epi.peaks %>% 
  as.data.frame() %>% 
  dplyr::rename(peak = ".") %>% 
  separate(peak, c("Chr", "start", "end"), sep = "-", remove = F) %>% 
  arrange(Chr, start, end) %>% 
  dplyr::filter(Chr == "Chr3") %>% 
  dplyr::filter(start > 69982446-2100) %>% 
  dplyr::filter(end < 69998194+500) %>% 
  inner_join(da_peaks_Epi %>% 
               mutate(peak = row.names(.)), by = "peak")
```
 
 


## Check STR peaks
```{r}
top.epi.peaks %>% 
  as.data.frame() %>% 
  dplyr::rename(peak = ".") %>% 
  separate(peak, c("Chr", "start", "end"), sep = "-", remove = F) %>% 
  arrange(Chr, start, end) %>% 
  dplyr::filter(Chr == "Chr3") %>% 
  dplyr::filter(start > 71678061-10000) %>% 
  dplyr::filter(end < 71679563+20000) %>% 
  inner_join(da_peaks_Epi %>% 
               mutate(peak = row.names(.)), by = "peak")
```


## Heatmap
```{r}
Average_ATAC <- AverageExpression(CRO_multiome, assay = "MACS2") %>% 
  as.data.frame() %>% 
  mutate(peak = row.names(.)) %>% 
  pivot_longer(cols = !peak, names_to = "cluster", values_to = "ATAC") %>%
  mutate(cluster = str_remove(cluster, "MACS2.")) %>% 
  mutate(cluster = factor(cluster, levels = c(
     "0", "1", "3", "7", 
    "2","5", 
    "12",
    "11", "6", "10", "14", "15",
    "13", "8",
    "4", "9"
  ))) %>% 
  mutate(cell_type = case_when(
    cluster == 0 | 
      cluster == 1 | 
      cluster == 3 | 
      cluster ==7  ~ "Mesophyll",
    cluster == 2 | 
      cluster == 5 ~ "Epidermis",
    cluster == 12 ~ "Guard cells",
    cluster == 11 |
      cluster == 6 |
      cluster == 10 |
      cluster == 14 |
      cluster == 15  ~ "Vasculature",
    cluster == 13 ~ "IPAP",
      cluster == 8 ~ "Idioblast",
    T ~ "Unassigned"
  )) %>% 
  mutate(cell_type = factor(cell_type, 
                            levels = c(
                              "Mesophyll", "Epidermis", "Guard cells",
                              "Vasculature", 
                              "IPAP", "Idioblast", "Unassigned"
                            ))) 

top_peaks <- Average_ATAC %>% 
  filter(peak %in% top.epi.peaks |
           peak %in% top.idioblat.peaks) %>% 
  mutate(class = case_when(
    peak %in% top.epi.peaks ~ "Epi peaks",
    peak %in% top.idioblat.peaks ~ "Id peaks"
  ))

head(top_peaks)
```

```{r}
top_peak_heatmap <- top_peaks %>% 
  mutate(ATAC_clipped = case_when(
    ATAC > quantile(ATAC, 0.9) ~ quantile(ATAC, 0.9),
    T ~ ATAC
  )) %>% 
  ggplot(aes(x = cluster, y = peak)) +
  facet_grid(class ~ cell_type, scales = "free", space = "free", switch = "y") +
  geom_tile(aes(fill = ATAC_clipped), color = NA) +
  scale_fill_gradientn(colors = carto_pal(7, "Sunset")) +
  labs(y = "peaks",
       x = "cluster",
       fill = "Normalized \nATAC signal ") +
  theme_classic() +
  theme(text = element_text(size = 14),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        legend.position = "top",
        legend.key.height = unit(0.8, "lines"),
        legend.key.width = unit(1.2, "lines"),
        strip.text.y.left = element_text(angle = 0, hjust = 0.5),
        strip.text.x = element_blank(),
        strip.background = element_blank()
        )

top_peak_heatmap
```


```{r}
Leaf_cell_type_strip_facet <- top_peaks %>% 
  ggplot(aes(x = cluster, y = "" )) +
  facet_grid(. ~ cell_type, scales = "free", space = "free", switch = "y") +
  geom_tile(aes(fill = cell_type)) +
  scale_fill_manual(values = c(brewer.pal(6, "Accent"), "grey80")) +
  labs(fill = "Cell type") +
  theme_void() +
  theme(
    legend.position = "bottom" ,
    text = element_text(size = 12),
    strip.text = element_blank()
  )

top_peak_heatmap_strip <- wrap_plots(top_peak_heatmap, Leaf_cell_type_strip_facet, 
           nrow = 2, heights = c(1, 0.07))

top_peak_heatmap_strip
ggsave("../Results/R_output/top_peak_heatmap.svg", height = 4.5, width = 6.5, bg = "white")
ggsave("../Results/R_output/top_peak_heatmap.png", height = 4.5, width = 6.5, bg = "white")
```


## Write out bed files for MEME
```{r}
top.idioblat.peaks %>% 
  as.data.frame() %>% 
  dplyr::rename(peak = ".") %>% 
  separate(peak, c("Chr", "start", "end"), sep = "-", remove = F) %>% 
  dplyr::select(Chr, start, end, peak) %>% 
  write_delim("../Results/R_output/top_idioblast_peaks.bed", delim = "\t", col_names = F)


top.epi.peaks %>% 
  as.data.frame() %>% 
  dplyr::rename(peak = ".") %>% 
  separate(peak, c("Chr", "start", "end"), sep = "-", remove = F) %>% 
  dplyr::select(Chr, start, end, peak) %>% 
  write_delim("../Results/R_output/top_epi_peaks.bed", delim = "\t", col_names = F)

top.IPAP.peaks %>% 
  as.data.frame() %>% 
  dplyr::rename(peak = ".") %>% 
  separate(peak, c("Chr", "start", "end"), sep = "-", remove = F) %>% 
  dplyr::select(Chr, start, end, peak) %>% 
  write_delim("../Results/R_output/top_IPAP_peaks.bed", delim = "\t", col_names = F)
```


## Write supp tables
```{r}
rbind(
  top.idioblat.peaks %>% 
  as.data.frame() %>% 
  mutate(attribute = "idioblast marker peak"),
  
  top.epi.peaks %>% 
  as.data.frame() %>% 
    mutate(attribute = "epidermis marker peak")
) %>% 
  dplyr::rename(peak = ".") %>% 
  separate(peak, c("Chr", "start", "end"), sep = "-", remove = F) %>% 
  dplyr::select(Chr, start, end, attribute)  %>% 
  arrange(attribute, Chr, start) %>% 
  write_excel_csv("../Results/R_output/epi_idio_marker_peaks.csv")

```


# Find overrepresented motifs 
```{r}
enriched.motifs.idioblast <- FindMotifs(
  object = CRO_multiome, 
  features = top.idioblat.peaks
) %>% 
  mutate(FDR = p.adjust(pvalue, method = "fdr")) %>%
  filter(pvalue < 0.01)

 enriched.motifs.epi <- FindMotifs(
  object = CRO_multiome, 
  features = top.epi.peaks
) %>% 
  mutate(FDR = p.adjust(pvalue, method = "fdr"))  %>% 
  filter(pvalue < 0.01)

enriched.motifs.idioblast
enriched.motifs.epi 
```

```{r}
Idioblast_motifs <- MotifPlot(
  object = CRO_multiome,
  motifs = enriched.motifs.idioblast$motif[c(1, 4)],
  nrow = 2
) &
  scale_y_continuous(breaks = c(0, 1, 2)) &
  theme(axis.text.x = element_blank(),
        axis.line.y = element_line(),
        axis.ticks.y = element_line()) 
  

Idioblast_motifs

ggsave("../Results/R_output/Idioblat_motifs.svg", height = 4, width = 3, bg = "white")
ggsave("../Results/R_output/Idioblat_motifs.png", height = 4, width = 3, bg = "white")
```

```{r}
wrap_plots(top_peak_heatmap_strip, Idioblast_motifs,
           design = c("A#B"), widths = c(1, 0.1, 0.35))

ggsave("../Results/R_output/top_peak_heatmap_id_motifs.svg", width = 7.7, height = 4, bg = "white")
ggsave("../Results/R_output/top_peak_heatmap_id_motifs.png", width = 7.7, height = 4, bg = "white")
```


```{r}
MotifPlot(
  object = CRO_multiome,
  motifs = c("MA1375.1", "MA0996.1", "MA1038.1"),
  ncol = 3
) &
  theme(axis.text.x = element_blank(),
        axis.line.y = element_line(),
        axis.ticks.y = element_line())

ggsave("../Results/R_output/Epidermis_motifs.svg", height = 2, width = 6, bg = "white")
ggsave("../Results/R_output/Epidermis_motifs.png", height = 2, width = 6, bg = "white")
```
```{r}
MotifPlot(
  object = CRO_multiome,
  motifs = c( "MA0996.1"),
  ncol = 1
) &
  theme(axis.text.x = element_blank(),
        axis.line.y = element_line(),
        axis.ticks.y = element_line()) 

ggsave("../Results/R_output/Epi_MA0996.svg", height = 2, width = 3, bg = "white")
ggsave("../Results/R_output/Epi_MA0996.png", height = 2, width = 3, bg = "white")
```
```{r}
MotifPlot(
  object = CRO_multiome,
  motifs = c( "MA0996.1","MA1238.2"),
  ncol = 1
) &
  theme(axis.text.x = element_blank(),
        axis.line.y = element_line(),
        axis.ticks.y = element_line()) 
```


# Save object 
```{r}
saveRDS(CRO_multiome, "../Results/R_output/CRO_multiome.Rds")
```



