---
title: "Mapping_metrics_dap_seq"
author: "Chenxin Li"
date: "2023-02-02"
output: html_notebook 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Packages 
```{r}
library(tidyverse)
library(RColorBrewer)
library(rcartocolor)
library(viridis)
library(patchwork)
```

# Data 
## MultiQC results 
```{r}
multiqc_stats <- read_delim("../Data/UGA.260123.AACK5M2M5_multiqc_30jan23_data/multiqc_general_stats.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE, col_types = cols())

head(multiqc_stats)
```

```{r}
multiqc_stats_bioreps <- multiqc_stats %>% 
  mutate(bio_rep = str_sub(Sample, end = 6)) %>% 
  group_by(bio_rep) %>% 
  summarise(
    total_SE_reads = sum(`FastQC_mqc-generalstats-fastqc-total_sequences`)
  ) %>% 
  ungroup() %>% 
  mutate(total_PE_reads = total_SE_reads/2)

head(multiqc_stats_bioreps)
```

## Total primary alignments 
```{r}
cro_bams <- read_table("../Results/cro_bam/cro_bam_files.txt"
                       , col_names = F, col_types = cols())


cro_pri_aln <- read_delim("../Results/cro_bam/cro_primary_aln.txt",
                          delim = " ", col_names = F, col_types = cols())

cro_bams <- cro_bams %>% 
  mutate(bio_rep = str_sub(X9, end = 6)) %>% 
  select(bio_rep) %>% 
  cbind(cro_pri_aln)  %>% 
  dplyr::rename(mapped_SE = X1) 

head(cro_bams)
```

```{r}
caa_bams <- read_delim("../Results/caa_bam/caa_bam_files.txt",
                       delim = " ", col_names = F, col_types = cols())

caa_pri_aln <- read_delim("../Results/caa_bam/caa_primary_aln.txt",
                          delim = " ", col_names = F, col_types = cols())

caa_bams <- caa_bams %>% 
  mutate(bio_rep = str_sub(X10, end = 6)) %>% 
  select(bio_rep) %>% 
  cbind(caa_pri_aln)  %>% 
  dplyr::rename(mapped_SE = X1)

head(caa_bams)
```

# Percentage reads mapped 
```{r}
multiqc_stats_bioreps_mapped <- multiqc_stats_bioreps %>% 
  full_join(rbind(
    cro_bams,
    caa_bams
  ), by = "bio_rep") %>% 
  mutate(percent_unique_mapped = mapped_SE / total_SE_reads * 100)

multiqc_stats_bioreps_mapped
```

# Number of reads remain after depublication 
```{r}
cro_dedup <- read_delim("../Results/cro_bam/cro_dedup.txt", 
                        delim = "\t", col_names = F, col_types = cols())

caa_dedup <- read_delim("../Results/caa_bam/caa_dedup.txt", 
                        delim = "\t", col_names = F, col_types = cols())
```

```{r}
multiqc_stats_bioreps_mapped_dedup <- multiqc_stats_bioreps_mapped %>% 
  cbind(
    rbind(
      caa_dedup,
      cro_dedup
    )
  ) %>% 
  mutate(unique_PE = X1/2)


multiqc_stats_bioreps_mapped_dedup
```

# Fraction of reads in peaks 
```{r}
peak_covs_list <- list.files("../Results/bedtools_out/", pattern = "*.cov", full.names = T)

peak_covs <- sapply(peak_covs_list, read_delim,
                     delim = "\t", col_names = F, col_types = cols(), simplify = F) %>% 
  bind_rows(.id = "id") %>% 
  mutate(bio_rep = case_when(
    str_detect(id, "CAA") ~ "CAA_BB",
    str_detect(id, "CRO_BH") ~ "CRO_BH",
    str_detect(id, "CRO_BI") ~ "CRO_BI")) 

peak_covs
```
```{r}
peak_cov_total <- peak_covs %>% 
  group_by(bio_rep) %>% 
  summarise(
    sum = sum(X4)
  ) %>% 
  inner_join(multiqc_stats_bioreps_mapped_dedup %>% 
               select(bio_rep, unique_PE), by = "bio_rep") %>% 
  mutate(FRiP = sum / unique_PE * 100 / 2)

peak_cov_total
```

# Extend ATAC and DAP peaks 
## Read ATAC
```{r}
ATAC_peaks <- read_delim("../../CRO_multiome/Results/MACS2_out/MACS2_peaks.bed",
                         delim = "\t", col_names = F, col_types = cols())

head(ATAC_peaks)
```
```{r}
ATAC_peaks_wide <- ATAC_peaks %>% 
  mutate(Chr = X1) %>% 
  mutate(start = X2 - 100) %>% 
  mutate(end = X3 + 100) %>% 
  mutate(start = case_when(
    start < 0 ~ 0,
    T ~ start 
  )) %>% 
  select(Chr, start, end) %>% 
  filter(str_detect(Chr, "Chr"))

head(ATAC_peaks_wide)
```

```{r}
peak_covs_wide <- peak_covs %>% 
  mutate(Chr = X1) %>% 
  mutate(start = X2 - 100) %>% 
  mutate(start = case_when(
    start < 0 ~ 0,
    T ~ start 
  )) %>% 
  mutate(end = X3 + 100) 

head(peak_covs_wide)
```

```{r}
peak_covs_wide %>% 
  filter(start < 0)
```


## Write bed files 
```{r}
write_delim(ATAC_peaks_wide, "../Results/R_outputs/ATAC_peaks_wide.bed", 
            delim = "\t", col_names = F)

write_delim(peak_covs_wide %>% 
              filter(bio_rep == "CAA_BB") %>%
              filter(str_detect(Chr, "Lachesis_group")) %>% 
              select(Chr, start, end), "../Results/R_outputs/CaORCA_DAP_peak_wide.bed", 
            delim = "\t", col_names = F)


write_delim(peak_covs_wide %>% 
              filter(bio_rep == "CRO_BH") %>% 
              filter(str_detect(Chr, "Chr")) %>% 
              select(Chr, start, end), "../Results/R_outputs/ORCA3_DAP_peak_wide.bed", 
            delim = "\t", col_names = F)


write_delim(peak_covs_wide %>% 
              filter(bio_rep == "CRO_BI") %>% 
              filter(str_detect(Chr, "Chr")) %>% 
              select(Chr, start, end), "../Results/R_outputs/ORCA4_DAP_peak_wide.bed", 
            delim = "\t", col_names = F)
```

# Size distribution
```{r}
Bocchi <- c(
  rgb(242, 201, 213, maxColorValue = 255), # bocchi
  rgb(180, 62, 68, maxColorValue = 255), # kita 
  rgb(250, 223, 146, maxColorValue = 255), # nijika
  rgb(73, 100, 150, maxColorValue = 255), # ryo 
  rgb(40, 40, 40, maxColorValue = 255), # PA
  rgb(144, 72, 105, maxColorValue = 255) # kikuri
  )
  
Bocchi
```

```{r}
ATAC_hist <- ATAC_peaks %>% 
  mutate(length = X3 - X2) %>% 
  ggplot(aes(x = length)) +
  geom_histogram(binwidth = 50, fill = Bocchi[2], color = "white") +
  scale_x_continuous(breaks = seq(0, 1500, by = 500)) +
  labs(x = "length (bp)",
       y = "num. of peaks") +
  theme_classic() +
  coord_cartesian(xlim = c(100, 1500)) +
  ggtitle("ATAC-seq")

CaORCA_hist <- peak_covs %>% 
  filter(bio_rep == "CAA_BB") %>% 
  mutate(length = X3 - X2) %>% 
  ggplot(aes(x = length)) +
  geom_histogram(binwidth = 50, fill = Bocchi[4], color = "white") +
  scale_x_continuous(breaks = seq(0, 1500, by = 500)) +
   labs(x = "length (bp)",
       y = "num. of peaks") +
  theme_classic() +
  coord_cartesian(xlim = c(100, 1500)) +
  ggtitle("CaORCA") 

ORCA3_hist <- peak_covs %>% 
  filter(bio_rep == "CRO_BH") %>% 
  mutate(length = X3 - X2) %>% 
  ggplot(aes(x = length)) +
  geom_histogram(binwidth = 50, fill = Bocchi[1], color = "black") +
   labs(x = "length (bp)",
       y = "num. of peaks") +
  scale_x_continuous(breaks = seq(0, 1500, by = 500)) +
  theme_classic() +
  coord_cartesian(xlim = c(100, 1500)) +
  ggtitle("CrORCA3")

ORCA4_hist <- peak_covs %>% 
  filter(bio_rep == "CRO_BI") %>% 
  mutate(length = X3 - X2) %>% 
  ggplot(aes(x = length)) +
  geom_histogram(binwidth = 50, fill = Bocchi[3], color = "black") +
  scale_x_continuous(breaks = seq(0, 1500, by = 500)) +
   labs(x = "length (bp)",
       y = "num. of peaks") +
  theme_classic() +
  coord_cartesian(xlim = c(100, 1500)) +
  ggtitle("CrORCA4") 

wrap_plots(ATAC_hist,
           CaORCA_hist,
           ORCA3_hist,
           ORCA4_hist) &
  labs(x = "length (bp)",
       y = "number of peaks") &
  theme(axis.text = element_text(color = "black"),
        title = element_text(size = 10))

ggsave("../Results/R_outputs/peak_length_hist.svg", height = 4, width = 5)
ggsave("../Results/R_outputs/peak_length_hist.png", height = 4, width = 5)
```
```{r}
peak_covs %>% 
  mutate(length = X3 - X2) %>%
  group_by(bio_rep) %>% 
  summarise(mean = mean(length),
            median = median(length))

ATAC_peaks_wide %>% 
  mutate(length = end - start) %>%
  #group_by(bio_rep) %>% 
  summarise(mean = mean(length),
            median = median(length))
```

# % accessible? 
```{r}
ORCA4_DAP_ATAC <- read_delim("../Results/bedtools_out/ORCA4_DAP_ATAC.bed",
                             delim = "\t", col_names = F, col_types = cols())

ORCA3_DAP_ATAC <- read_delim("../Results/bedtools_out/ORCA3_DAP_ATAC.bed",
                             delim = "\t", col_names = F, col_types = cols())

num_accessible <- rbind(
  ORCA3_DAP_ATAC %>% 
    mutate(bio_rep = "CRO_BH"),
  ORCA4_DAP_ATAC %>% 
    mutate(bio_rep = "CRO_BI")
) %>% 
  group_by(bio_rep) %>% 
  dplyr::count() %>%
  ungroup() %>%
  dplyr::rename(accessible_peaks = n) %>% 
  inner_join(
    peak_covs %>% 
      group_by(bio_rep) %>% 
      dplyr::count() %>% 
      ungroup()
  ) %>% 
  dplyr::rename(total_peaks = n) %>% 
  mutate(inaccessible_peaks = total_peaks - accessible_peaks) %>% 
  mutate(percent_accessible = accessible_peaks/total_peaks * 100)

num_accessible
```
```{r}
accessible_stacked_bar <-num_accessible %>% 
  mutate(percent_in = 100 - percent_accessible) %>% 
  mutate(TF = case_when(
    bio_rep == "CRO_BH" ~ "ORCA3",
    bio_rep == "CRO_BI" ~ "ORCA4"
  )) %>% 
  pivot_longer(cols = c(percent_accessible, percent_in), 
               names_to = "type", values_to = "percent") %>% 
  mutate(Type = case_when(
    str_detect(type, "acces") ~ "accessible",
    T ~ "inaccessible"
  )) %>% 
  ggplot(aes(x = TF, y = percent)) +
  geom_bar(stat = "identity", aes(fill = Type), color = "white", 
           alpha = 0.8, width = 0.7) +
  scale_fill_manual(values = Bocchi[c(2,5)]) +
  guides(fill = guide_legend(ncol = 1)) +
  labs(x = "DAP-seq peaks",
       y = "%") +
  theme_classic() +
  theme(axis.text = element_text(color = "black"),
        legend.position = "right")

accessible_stacked_bar
ggsave("../Results/R_outputs/percent_accessible.svg", height = 3, width = 2, bg = "white")
ggsave("../Results/R_outputs/percent_accessible.png", height = 3, width = 2, bg = "white")
```

```{r}
wrap_plots(ATAC_hist,
           ORCA3_hist,
           ORCA4_hist,
           accessible_stacked_bar) +
  theme(axis.text = element_text(color = "black"),
        title = element_text(size = 10)) 

ggsave("../Results/R_outputs/length_dist2.svg", height = 4.5, width = 6, bg = "white")
ggsave("../Results/R_outputs/length_dist2.png", height = 4.5, width = 6, bg = "white")
```


# Location of peaks?
```{r}
nearest.list <- list.files("../Results/bedtools_out/",
                           pattern = "*.txt", full.names = T)

nearest.data <- sapply(nearest.list, read_delim, delim = "\t",
                       col_names = F, col_types = cols(), simplify = F) %>% 
  bind_rows(.id = "id") %>% 
  mutate(TF = case_when(
    str_detect(id, "CaORCA") ~ "CaORCA",
    str_detect(id, "ORCA3") ~ "ORCA3",
    str_detect(id, "ORCA4") ~ "ORCA4"
  )) %>% 
  mutate(type = case_when(
    str_detect(id, "all") ~ "all peaks",
    T ~ "accessible peaks"
  ))

head(nearest.data)

nearest.data %>% 
  group_by(TF, type) %>% 
  count()
```
## Distance to genes? 
```{r}
nearest.data_trim <- nearest.data %>% 
  mutate(peak = paste0(X1, ":", X2, "-", X3)) %>% 
  group_by(peak) %>% 
  slice_min(order_by = X8) %>% 
  ungroup() 

head(nearest.data_trim)

nearest.data_trim %>% 
  group_by(TF, type) %>% 
  count()
```
```{r}
nearest.data_trim %>% 
  ggplot(aes(x = TF, y = X8+1)) +
  facet_grid(. ~ type, scales = "free", space = "free_x") +
  geom_violin() +
  stat_summary(geom = "point", aes(fill = TF), 
               fun.data = mean_se, size = 3, color = "black", shape = 21) +
  scale_y_log10(breaks = c(10, 100, 10^2, 10^3, 10^4, 10^5)) +
  scale_fill_manual(values = Bocchi[c(4, 1, 3)]) +
  labs(y = "distance to nearest gene (bp)",
       x = "DAP-seq peaks") +
  theme_classic() +
  theme(legend.position = "bottom",
        axis.text.x = element_blank())

ggsave("../Results/R_outputs/peak_gene_dist.svg", height = 3, width = 5, bg = "white")
ggsave("../Results/R_outputs/peak_gene_dist.png", height = 3, width = 5, bg = "white")
```

## Classification by distance relative to genes  
```{r}
nearest.data_trim_2 <- nearest.data_trim %>% 
  mutate(class = case_when(
    X8 == 0 ~ "overlap a gene",
    X8 > 0 & X8 <= 500 ~ "proximal (0-500)",
    X8 > 500 & X8 <= 2000 ~ "500-2000",
    X8 > 2000 ~ "distal"
  )) %>% 
  group_by(TF, type, class) %>% 
  count() %>% 
  ungroup() %>% 
  mutate(class = factor(class, levels = c(
    "overlap a gene",
    "proximal (0-500)",
    "500-2000",
    "distal"
  ))) %>%
  rename(num.peak = n) %>% 
  inner_join(
    nearest.data_trim %>% 
  group_by(TF, type) %>% 
  count(), 
  by = c("TF", "type") 
  ) %>% 
  mutate(percent = num.peak / n * 100)

nearest.data_trim_2
```

```{r}
nearest.data_trim_2 %>% 
  ggplot(aes(x = TF, y = percent)) +
  facet_grid(. ~ type, scales = "free", space = "free_x") +
  geom_bar(stat = "identity", aes(fill = class), 
           alpha = 0.8, width = 0.5) +
  scale_fill_manual(values = rev(viridis(10, option = "D")[c(1, 3, 7, 10)])) +
  guides(fill = guide_legend(ncol = 2)) +
  labs(x = "DAP-seq peaks",
       y = "%",
       fill = "location") +
  theme_classic() +
  theme(axis.text = element_text(color = "black"),
        legend.position = "top")

ggsave("../Results/R_outputs/peak_gene_dist2.svg", height = 3, width = 5, bg = "white")
ggsave("../Results/R_outputs/peak_gene_dist2.png", height = 3, width = 5, bg = "white")
```


# CAA genes and coordinates 
```{r}
Kang_genes <- readxl::read_excel("../Data/Kang2021_MEP_iridoid.xlsx")
head(Kang_genes)
```

```{r}
CAA_gff3 <- read_delim("../Data/CAA/genome.HiC.gff3",
                       delim = "\t", col_names = F, col_types = cols())

head(CAA_gff3)
```
```{r}
Kang_genes_nice <- CAA_gff3 %>% 
  filter(X3 == "gene") %>% 
  separate(X9, c("Gene_ID", "Name"), sep = ";") %>%
  select(-Name) %>% 
  mutate(Gene_ID = str_remove(Gene_ID, "ID=")) %>% 
  right_join(Kang_genes, by = "Gene_ID") %>% 
  arrange(Order)

head(Kang_genes_nice)
```
```{r}
write_delim(CAA_gff3 %>% 
              filter(str_detect(X1, "Lachesis")) %>% 
  filter(X3 == "gene") %>% 
    arrange(X1, X4), delim = "\t", col_names = F,
  "../Results/R_outputs/CAA_genes.gff3")
```

```{r}
write_excel_csv(Kang_genes_nice, "../Results/R_outputs/Kang_genes_nice.csv")
```

